src/main/java/com/artipie/http/PromuSlice.java

```diff
@@ -51,41 +53,51 @@ public PromuSlice(final Metrics metrics) {
-        final StringWriter writer = new StringWriter();
-        final Set<String> filters = PromuSlice.params(line, "name");
-        final MetricsOutput output = new PrometheusOutput(writer, ctype, filters);
-        this.metrics.publish(output);
-        return new RsFull(
-            RsStatus.OK,
-            new Headers.From(new ContentType(ctype)),
-            new Content.From(
-                String.valueOf(writer.getBuffer()).getBytes(StandardCharsets.UTF_8)
-            )
-        );
+        final String ctype = new RqHeaders(headers, Accept.NAME).get(0);
+        try (StringWriter writer = new StringWriter()) {
+            final Set<String> filters = PromuSlice.params(line, "name");
+            final MetricsOutput output = new PrometheusOutput(writer, ctype, filters);
+            this.metrics.publish(output);
+            return new RsFull(
+                RsStatus.OK,
+                new Headers.From(new ContentType(ctype)),
+                new Content.From(
+                    String.valueOf(writer.getBuffer()).getBytes(StandardCharsets.UTF_8)
+                )
+            );
+        } catch (final IOException ioe) {
+            throw new ArtipieIOException(ioe);
+        }
     }
 
     /**
      * Get all params with name in query.
-     * @param query Query
+     * @param line Line
      * @param name Name
      * @return All params with this name
      */
-    private static Set<String> params(final String query, final String name) {
-        return StreamSupport.stream(
-            Splitter.on("&").omitEmptyStrings().split(query).spliterator(),
-            false
-        ).flatMap(
-            param -> {
-                final String prefix = String.format("%s=", name);
-                final Stream<String> value;
-                if (param.startsWith(prefix)) {
-                    value = Stream.of(param.substring(prefix.length()));
-                } else {
-                    value = Stream.empty();
+    private static Set<String> params(final String line, final String name) {
+        final Set<String> values;
+        final String query = new RequestLineFrom(line).uri().getQuery();
+        if (query == null) {
+            values = new HashSet<>();
+        } else {
+            values = StreamSupport.stream(
+                Splitter.on("&").omitEmptyStrings().split(query).spliterator(),
+                false
+            ).flatMap(
+                param -> {
+                    final String prefix = String.format("%s=", name);
+                    final Stream<String> value;
+                    if (param.startsWith(prefix)) {
+                        value = Stream.of(param.substring(prefix.length()));
+                    } else {
+                        value = Stream.empty();
+                    }
+                    return value;
                 }
-                return value;
-            }
-        ).collect(Collectors.toSet());
+            ).collect(Collectors.toSet());
+        }
```

Comment:
You can use this object instead of new code: https://github.com/artipie/http/blob/master/src/main/java/com/artipie/http/rq/RqParams.java

Reply:
It only gives us the first value of the name. We want all values. That's why I reported it [here](https://github.com/artipie/http/issues/412) in `artipie/http`.

Reply:
It only gives us the first value of the name. We want all values. That's why I reported it [here](https://github.com/artipie/http/issues/412) in `artipie/http`.

Reply:
Thanks, reported here: https://github.com/artipie/http/issues/413

Reply:
Oh, sorry, I just realized you've already reported it...